From e31c1971697e2dce92b86a520809b42421b51852 Mon Sep 17 00:00:00 2001
From: Trung LE <trung.le@ruby-journal.com>
Date: Mon, 9 Mar 2020 03:37:56 +0000
Subject: [PATCH] Support custom gn binary for ppc64le platform

---
 gn.py | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/gn.py b/gn.py
index 43b18dd0..175cfe64 100755
--- a/gn.py
+++ b/gn.py
@@ -18,7 +18,7 @@ import gclient_paths
 import os
 import subprocess
 import sys
-
+import platform

 def PruneVirtualEnv():
   # Set by VirtualEnv, no need to keep it.
@@ -54,9 +54,14 @@ def main(args):
     if os.path.exists(gn_path):
       return subprocess.call([gn_path] + args[1:])

-  # Otherwise try the old .sha1 and download_from_google_storage locations
-  # inside of buildtools.
-  bin_path = gclient_paths.GetBuildtoolsPlatformBinaryPath()
+  # Fallback to custom-built gn for linux ppc64le
+  if platform.machine() == 'ppc64le':
+    bin_path = os.environ['GN_BIN_PATH']
+  else:
+    # Otherwise try the old .sha1 and download_from_google_storage locations
+    # inside of buildtools.
+    bin_path = gclient_paths.GetBuildtoolsPlatformBinaryPath()
+
   if not bin_path:
     print('gn.py: Could not find checkout in any parent of the current path.\n'
           'This must be run inside a checkout.', file=sys.stderr)
--
2.17.1
